MoltenVK/MoltenVK/GPUObjects/MVKVulkanAPIObject.h

```diff
@@ -126,7 +126,8 @@ class MVKDispatchableVulkanAPIObject : public MVKVulkanAPIObject {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wobjc-method-access"
 /** Generically avoids setting a label to nil, which many objects don't like. */
-static inline void setLabelIfNotNil(id object, NSString* label) { if (label) { [object setLabel: label]; } }
+extern std::atomic<uint64_t> nextMetalObjectIdentifier;
+static inline void setLabelIfNotNil(id object, NSString* label) { if (label) { [object setLabel: [label stringByAppendingFormat:@" (%llu)", nextMetalObjectIdentifier++]]; } }
```

Comment:
1. Is the use of an atomic counter specifically necessary? Could we just `%p` the object address as it's identifier in the debug string?
2. `setLabelIfNotNil()` is called in some very performance-sensitive areas, and is often attached to transient or pooled Metal objects. Because of this, we almost always called it with with a fixed (and hence cached) `NSString*`instance. It would be good if we could avoid creating a dynamic string allocation in these areas, unless we're in Debug mode. Since this is a static function, we can either guard the dynamic string with an `#if MVK_DEBUG` macro, or `if(getGlobalMVKConfig().debugMode) {}`. Ultimately, we should move this function to `MVKBasicObject`, where we could call `getMVKConfig()`, to allow layer setting config support.